kore/src/Kore/Builtin/InternalBytes.hs

```diff
@@ -214,6 +227,46 @@ evalString2Bytes =
             & return
     evalString2Bytes0 _ _ _ = Builtin.wrongArity string2BytesKey
 
+evalDecodeBytes :: BuiltinAndAxiomSimplifier
+evalDecodeBytes = Builtin.applicationEvaluator evalDecodeBytes0
+  where
+    evalDecodeBytes0 _ app
+        | [_strTerm, _bytesTerm] <- applicationChildren app = do
+            let Application{applicationSymbolOrAlias = symbol} = app
+                resultSort = symbolSorts symbol & applicationSortsResult
+                returnResult = return . String.asPattern resultSort
+            _str <- String.expectBuiltinString decodeBytesKey _strTerm
+            _bytes <- matchBuiltinBytes _bytesTerm
+            case Text.unpack _str of
+                "UTF-8" -> case Text.decodeUtf8' _bytes of
+                    Right str -> returnResult str
+                    Left _ -> return (Pattern.bottomOf resultSort) -- UnicodeException
```

Comment:
Wouldn't it be surprising to have a `UTF-8` unsuccessful decoding returning `bottom` but the other ones throwing exceptions? Maybe this case should be documented?

Reply:
Good point, I will add documentation to hooks.md describing this behavior.